src/main/kotlin/app/ui/ConsoleUi.kt

```diff
@@ -0,0 +1,20 @@
+// Copyright 2017 Sourcerer Inc. All Rights Reserved.
+// Author: Anatoly Kislov (anatoly@sourcerer.io)
+
+package app.ui
+
+/**
+ * Console user interface.
+ */
+class ConsoleUi : Context {
+    var state: ConsoleState = OpenState(this)
+
+    init {
+        state.doAction()
+    }
+
+    override fun changeState(state: ConsoleState) {
+        this.state = state
+        state.doAction()
```

Comment:
If you had next() as a method of ConsoleState, you could call it right after doAction()